arduino/smartcar/smartcar.ino

```diff
@@ -151,18 +153,32 @@ void sendObstacleDetectedNotification(boolean shouldSend){
 }  
 
 void autonomousMoving(){
-  car.setSpeed(stoppingSpeed);
-        delay(1000);
-        car.setSpeed(-autoSpeed);
-        delay(1000);
         car.setSpeed(stoppingSpeed);
-        delay(1000);
-        car.setSpeed(autoSpeed);
-        car.setAngle(-autoAngle);
-        delay(1000);
+        delay(500);
+        car.setSpeed(-autoSpeed);
+        delay(500);
         car.setSpeed(stoppingSpeed);
-        car.setAngle(stopAngle);
-        delay(1000);
+        delay(500);
+        turning(turnLeft);
+        auto frontDistanceFromObject = frontSensorUS.getDistance();
+        boolean isFrontDetected = frontDistanceFromObject < stopDistanceFront && frontDistanceFromObject > 1 && !(carSpeed <= 0);
+        if (isFrontDetected){
+          turning(turnRight);
+          turning(turnRight);
+          frontDistanceFromObject = frontSensorUS.getDistance();
+          isFrontDetected = frontDistanceFromObject < stopDistanceFront && frontDistanceFromObject > 1 && !(carSpeed <= 0);
+          if (isFrontDetected){
```

Comment:
These lines can then be changed to:
```suggestion
if (obstacleAvoidance()) {
    turning(turnRight);
    turning(turnRight);
    if (obstacleAvoidance()) {
```

This would reuse the previous obstacleAvoidance() code, and also not need to duplicate the code from that method.